### PR TITLE
[distribution_list][CHG] Improve performance

### PR DESCRIPTION
**Pull request by JonathanNEMRY**
Avoid doing redundant treatment into the loop case where we are in `safe_mode`

### CODE REVIEW

File: distribution_list/distribution_list.py [hunk 1]
```diff
@@ -41,6 +41,33 @@ class distribution_list(orm.Model):
 
     _name = 'distribution.list'
 
+    def _get_computed_ids(self, cr, uid, distribution_list, computed_ids,
```
Comment (Olivier-LAURENT) [inline distribution_list/distribution_list.py]: Do not pass computed_ids nor distribution_list, pass only the bridge field

File: distribution_list/distribution_list.py [hunk 2]
```diff
@@ -41,6 +41,33 @@ class distribution_list(orm.Model):
...
+                          to_be_computed_ids, context=None):
+        """
+        case where there are multiple models: ids of dst_model are 'into'
```
Comment (Olivier-LAURENT) [inline distribution_list/distribution_list.py]: Convert source ids to target ids according to the bridge field

File: distribution_list/distribution_list.py [hunk 3]
```diff
@@ -41,6 +41,33 @@ class distribution_list(orm.Model):
...
+                    if r and r.get(distribution_list.bridge_field, False):
+                        if isinstance(r[distribution_list.bridge_field],
+                                      tuple):
+                            computed_ids.append(
+                                r[distribution_list.bridge_field][0])
+                        else:
+                            computed_ids.append(
+                                r[distribution_list.bridge_field])
+        else:
+            include_temp = to_be_computed_ids.values()
```
Comment (Olivier-LAURENT) [inline distribution_list/distribution_list.py]: Loop on all values (better intuitive)

File: distribution_list/distribution_list.py [hunk 4]
```diff
@@ -41,6 +41,33 @@ class distribution_list(orm.Model):
...
+            if include_temp:
+                computed_ids += include_temp[0]
+        return computed_ids
```
Comment (Olivier-LAURENT) [inline distribution_list/distribution_list.py]: return the set() of result ids

**Comment by Cedric-Pigeon** [comment]
:+1:

**Comment by adrienpeiffer** [comment]
:+1:

**Pull request merged**